interp/type.go

```diff
@@ -854,8 +854,14 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		if err != nil {
 			return nil, err
 		}
-		if t.cat == ptrT {
+		switch t.cat {
+		case ptrT:
 			t = t.val
+		case valueT:
+			switch t.rtype.Kind() {
+			case reflect.Array, reflect.Ptr:
+				t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
+			}
 		}
 		if t.cat == arrayT {
```

Comment:
this should be moved as a case within the newly introduced switch L857, no?

Reply:
nevermind, I was confused by the succesion of switch and if. both indeed need to be done.